language/stdlib/modules/Roles.move

```diff
@@ -295,6 +295,48 @@ module Roles {
         move_to(new_account, Privilege<UnhostedRole>{ witness: UnhostedRole{}, is_extracted: false })
     }
 
+    // DD refactor: privilege-checking functions for roles
+    /// has_*_role and has_*_privilege functions are derived exactly from the
+    /// "Libra Governance roles & permissions" spreadsheet
+    /// https://docs.google.com/spreadsheets/d/13pL_0JCdNkGTMemVYKLvGifiF-svP_fW_pT6CjxFMCA/edit#gid=0
+    /// MintCurrency, BurnCurrency, PreburnCurrency, RotateAuthenticationKey, WithdrawalCapability are
+    /// omitted because they all involve real capabilities. They were also not handled in the
+    /// Roles module before refactoring.
+    ///
+    /// Naming conventions: Many of the "has_*_privilege" functions do have the same body
+    /// because the spreadsheet grants all such privileges to addresses (usually a single
+    /// address) with that role. In effect, having the privilege is equivalent to having the
+    /// role, but the function names document the specific privilege involved.  Also, modules
+    /// that use these functions as a privilege check can hide specific roles, so that a change
+    /// in the privilege/role relationship can be implemented by changing Roles and not the
+    /// module that uses it.
+    public fun has_treasury_compliance_role(tc_account: &signer): bool acquires RoleId {
+       let tc_address = Signer::address_of(tc_account);
+            exists<RoleId>(tc_address)
```

Comment:
nit: please do not indent after let. The expression on the next line after let has the same indentation as the let. The style you are using is super uncommon.

Reply:
Mistake. Fixed.